runtime/runtime/src/actions.rs

```diff
@@ -699,6 +701,33 @@ pub(crate) fn action_use_global_contract(
         GlobalContractIdentifier::CodeHash(code_hash) => AccountContract::Global(*code_hash),
         GlobalContractIdentifier::AccountId(id) => AccountContract::GlobalByAccount(id.clone()),
     };
+    let storage_cost = apply_state
+        .config
+        .fees
+        .storage_usage_config
+        .global_contract_usage_storage_amount_per_identifier_byte
+        .saturating_mul(action.contract_identifier.len() as u128);
+    let Some(updated_balance) = account.amount().checked_sub(storage_cost) else {
+        result.result = Err(ActionErrorKind::LackBalanceForState {
+            account_id: account_id.clone(),
+            amount: storage_cost,
+        }
+        .into());
+        return Ok(());
+    };
+    stats.balance.global_actions_burnt_amount =
+        stats.balance.global_actions_burnt_amount.saturating_add(storage_cost);
```

Comment:
this is not needed, we should not burn any tokens as part of this action
it is enough to update `storage_usage`, that will automatically lock user tokens
it also means that `global_contract_usage_storage_amount_per_identifier_byte` is not needed, please remove it